pkg/nsx/client.go

```diff
@@ -83,5 +100,29 @@ func GetClient(cf *config.NSXOperatorConfig) *Client {
 		SecurityClient: securityClient,
 		RuleClient:     ruleClient,
 		NSXChecker:     *nsxChecker,
+		NSXVerChecker:  *nsxVersionChecker,
+	}
+}
+
+func (client *Client) CheckNsxVersion() bool {
+	if client.NSXVerChecker.versionSupported {
+		return true
+	}
+
+	nsxVersion, err := client.NSXVerChecker.cluster.GetVersion()
+	if err != nil {
+		log.Error(err, "get version error")
+		return false
+	}
+	err = nsxVersion.Validate()
+	if err != nil {
+		log.Error(err, "validate version error")
+		return false
+	}
+
+	if !nsxVersion.featureSupported(minVersion) {
```

Comment:
Not related to your patch but I doubt that if the func `featureSupported` can work as expected... What cases did you test? Did you verify it with real NSX or mocked NSX?